### ISSUE TITLE
docgen: do we need separate `annex` pages?

### ISSUE THREAD

**Issue by karel-3d** [gunk/gunk#299]
I don't understand fully what do we need `annex.md` pages for in docgen.

It includes messages and enums that are all included in the service pages. (I am now fixing so in the annex section we see correct enums used in the messages.)

When we have that fixed, do we still need `annex.md` pages for types?

cc @favadi , @rnd

**Comment by karel-3d** [comment]
what I am proposing is just not generating anything with docgen if the file has no services.

We currently have
```
tplName := "annex.md"
if api.HasServices() {
	tplName = "api.md"
}
```

but instead, we could just not generate anything... I am not fully sure how to do that in the current setup ("skip" the file) and what would it break in the pipeline though

**Comment by karel-3d** [comment]
solved by #300 and #302
